### PR TITLE
defaults boost.context on riscv64

### PR DESCRIPTION
**Pull request by ct-clmsn**
## Fixes #

N/A

## Proposed Changes

  - when riscv64 detected, automatically sets `HPX_WITH_GENERIC_CONTEXT_COROUTINES=ON` (Boost.Context)
 
## Any background context you want to provide?

- eases compilation process for users; "quality of life" modification.

## Checklist

Not all points below apply to all pull requests.

- [x] I have added a new feature and have added tests to go along with it.
- [ ] I have fixed a bug and have added a regression test.
- [ ] I have added a test using random numbers; I have made sure it uses a seed, and that random numbers generated are valid inputs for the tests.

### CODE REVIEW

File: cmake/TargetArch.cmake [hunk 1]
```diff
@@ -146,6 +146,15 @@ function(target_architecture output_var)
...
       string(REPLACE "cmake_ARCH " "" ARCH "${ARCH}")
 
+      # If compiling for riscv64, automatically bake in Boost.Context
+      if(ARCH STREQUAL "riscv64")
+        hpx_set_option(
+            HPX_WITH_GENERIC_CONTEXT_COROUTINES
+            VALUE ON
+            FORCE
+        )
+      endif()
```
Comment (hkaiser) [inline cmake/TargetArch.cmake:156]: May I suggest adding this option here instead: https://github.com/STEllAR-GROUP/hpx/blob/d26d4126260d93ff9a4ddad877e7ee27c115ed9e/CMakeLists.txt#L1564-L1577?

Comment (ct-clmsn) [reply cmake/TargetArch.cmake]: all set! had to move target arch detection a little earlier.

Comment (hkaiser) [reply cmake/TargetArch.cmake]: Thanks!

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**